### PR TITLE
Add JUnit coroutine test rule

### PR DESCRIPTION
**Pull request by NotWoods**
Hopefully this makes it simpler to override Dispatchers.Main for testing. Based loosely on https://craigrussell.io/2019/11/unit-testing-coroutine-suspend-functions-using-testcoroutinedispatcher/ and Android Dev Summit slides.

---

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- [ ] **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/browser/awesomebar/src/test/java/mozilla/utils/coroutines.kt [hunk 1]
```diff
@@ -1,35 +0,0 @@
...
-import kotlinx.coroutines.ExperimentalCoroutinesApi
-import kotlinx.coroutines.asCoroutineDispatcher
-import kotlinx.coroutines.test.resetMain
-import kotlinx.coroutines.test.setMain
-import java.util.concurrent.Executors
-
-/**
- * Create single threaded dispatcher for test environment.
- */
-fun createTestCoroutinesDispatcher(): CoroutineDispatcher =
```
Comment (Amejia481) [inline components/browser/awesomebar/src/test/java/mozilla/utils/coroutines.kt]: As this is public API, I think we should follow [the deprecating guideline](https://mozac.org/contributing/deprecating). What do you think?

Comment (NotWoods) [reply components/browser/awesomebar/src/test/java/mozilla/utils/coroutines.kt]: Is this a public API? I thought it was just a testing helper in the awesomebar component.

Comment (Amejia481) [reply components/browser/awesomebar/src/test/java/mozilla/utils/coroutines.kt]: Oh, sorry my bad you are right!

**Comment by NotWoods** [comment]
bors r=Amejia481

**Pull request merged**